python/tests/test_nn.py

```diff
@@ -8,6 +8,7 @@
 import mlx.nn as nn
 import mlx_tests
 import numpy as np
+import torch
```

Comment:
We prefer not to have a non-optional dependency on `torch` in the test suite. Could you remove this and update the tests accordingly?

Reply:
FWIW we have found bugs in MPS in the past (some of which have not yet been resolved) so I would also not trust the MPS back-end as the source of truth. We should have tests which we have more confidence in.

Reply:
removed, but still kept the helper in mx to check shape/dtype/correctness